base/deprecated.jl

```diff
@@ -1758,4 +1758,13 @@ export @test_approx_eq
 @deprecate(SharedArray{T}(filename::AbstractString, ::Type{T}, dims::NTuple, offset; kwargs...),
            SharedArray{T,length(dims)}(filename, dims, offset; kwargs...))
 
+# limit use of LibGit2.get method (part of #19839)
+eval(Base.LibGit2, quote
+     @deprecate get{T<:GitObject}(::Type{T}, repo::GitRepo, x) T(repo, x)
+     @deprecate get{T<:GitObject}(::Type{T}, repo::GitRepo, oid::GitHash, oid_size::Int) T(repo, GitShortHash(oid, oid_size))
+     @deprecate revparse(repo::GitRepo, objname::AbstractString) GitObject(repo, objname)
+     @deprecate object(repo::GitRepo, te::GitTreeEntry) GitObject(repo, te)
+     @deprecate commit(ann::GitAnnotated) GitHash(ann)
```

Comment:
These are sometimes used externally (e.g. PkgDev calls LibGit2.get).
I've added an option to `@deprecate` on whether to export or not.